src/shogun/regression/LeastAngleRegression.cpp

```diff
@@ -341,18 +341,22 @@ bool CLeastAngleRegression::train_machine_templated(CDenseFeatures<ST>* data)
 			va.push_back((float64_t) m_beta_path_t[i][p]);			
 		}
 		m_beta_path.push_back(va);
+
+		SGVector<float64_t> obs_beta_path(va.size());
+		std::copy(va.begin(), va.end(), obs_beta_path.vector);
+		observe(i, "beta_path", "Beta path", obs_beta_path);
```

Comment:
`va` has the new values concatenated in every iteration, i.e. the old values still stay the same. So if we observe them, the last observation is sufficient to have all elements in the path or?
Then the question is: why do we even need to observe them? Or am I lost (again)? ;)

Reply:
My idea was to expose the different betas value such to produce this kind of visualizations https://en.wikipedia.org/wiki/Least-angle_regression#/media/File:Larsdiabetes.png. One could still do it by using the method `get_w_for_var()`, though.

Reply:
Ah I got confused, the betas are stored in `m_beta_path`, not in `va` (which just contains the last one).

OK then.

Playing devils advocat here: in order to produce the visualization, all you would need is `m_beta_path` at the end (contains all betas). So storing those betas inside the observer doesnt really add anything.

Reply:
Oh one thing: I dont really like the `std::copy` inside here. I think the copy should happen inside the observer that wants to store values. Less redundant. Is that possible?

Reply:
> Ah I got confused, the betas are stored in `m_beta_path`, not in `va` (which just contains the last one).
> 
> OK then.
> 
> Playing devils advocat here: in order to produce the visualization, all you would need is `m_beta_path` at the end (contains all betas). So storing those betas inside the observer doesnt really add anything.

That's true, but you need to wait until the training is finished to read the complete `m_beta_path`. I would like to give to the user the ability to write "interactive" models. That means that I could write a script which trains LARS and that shows me a plot with the LARS paths in real-time (while the machine is training). This could be achieved by providing an observer which executes custom code given by the user (like a lambda). However, if LARS does not emit the paths while it is fitted then this kind of things are not possible.

Reply:
> Oh one thing: I dont really like the `std::copy` inside here. I think the copy should happen inside the observer that wants to store values. Less redundant. Is that possible?

I had to do it since`va` is a `std::vector` and it is not recognized by the `sg_any_dispatch`. I agree when you say that it's redundant though. Let me see if I can find a better solution.

Reply:
aaaah
@gf712 is there something we can maybe do about this?

Reply:
ok that seems addressed.
RE the real time, I like the idea and think it is good
But maybe write about this somewhere (meta example, blog post, etc)

Reply:
This code is not needed anymore since LARS works natively with `SGVector` now.